include/picongpu/fields/incidentField/profiles/InsightPulse.hpp

```diff
@@ -0,0 +1,618 @@
+
+                            // starting with aligning the second transversal direction, perp. to polarisation and
+                            // propagation
+                            DataSpace<3u> aligningAxisIndex
+                                = DataSpace<3u>::create(static_cast<int>(pmacc::math::abs(pmacc::math::dot(
+                                    xyzAxisIndex,
+                                    pmacc::math::cross(Functor::getDirection(), Functor::getPolarisationVector())))));
+
+                            // aligning propagation direction
+                            int const propAxisIdx = static_cast<int>(
+                                pmacc::math::abs(pmacc::math::dot(xyzAxisIndex, Functor::getDirection())));
+                            auto it_prop
+                                = std::find(axisLabels.begin(), axisLabels.end(), Params::propagationAxisOpenPMD);
+                            if(it_prop != std::end(axisLabels))
+                                aligningAxisIndex[std::distance(begin(axisLabels), it_prop)] = propAxisIdx;
+                            else
+                                throw std::runtime_error(
+                                    "Error: could not find propagation axis "
+                                    + std::string(Params::propagationAxisOpenPMD) + " in OpenPMD dataset");
+
+                            // aligning polarisation direction
+                            auto it_pola
+                                = std::find(axisLabels.begin(), axisLabels.end(), Params::polarisationAxisOpenPMD);
+                            if(it_pola != std::end(axisLabels))
+                                aligningAxisIndex[std::distance(begin(axisLabels), it_pola)]
+                                    = static_cast<int>(pmacc::math::abs(
+                                        pmacc::math::dot(xyzAxisIndex, Functor::getPolarisationVector())));
+                            else
+                                throw std::runtime_error(
+                                    "Could not find polarisation axis " + std::string(Params::polarisationAxisOpenPMD)
+                                    + " in OpenPMD dataset");
+
+                            ::openPMD::MeshRecordComponent meshRecord = mesh[Params::polarisationAxisOpenPMD];
+
+                            //! necessary attributes
+                            // Raw = not yet aligned
+                            ::openPMD::Extent const extentRaw = meshRecord.getExtent();
+                            auto const cellSizeRaw = mesh.gridSpacing<dataType>();
+
+                            bufferExtentOpenPMD = std::make_shared<pmacc::HostDeviceBuffer<float_X, 1u>>(3u);
+                            bufferCellSizeOpenPMD = std::make_shared<pmacc::HostDeviceBuffer<float_X, 1u>>(3u);
+                            bufferOffsetOpenPMD = std::make_shared<pmacc::HostDeviceBuffer<float_X, 1u>>(3u);
+
```

Comment:
I'm not so familiar with the concept of shared pointers in C++. If there is another way to solve this, please let me know - the fact is that I need 3 float-vectors of length 3 to describe the field data chunk completely (extent, cell size and the offset w.r.t. the simulation window). I calculate those values at the same timestep when the data is loadad  and since I need them outside the openPMD-singleton, I put them in a HostDeviceBuffer (just like the corresponding field data chunk).